### PR TITLE
feat(kuma-cp): add possibility to configure kubernetes client qps and…

### PR DESCRIPTION
**Pull request by Automaat**
… burst qps

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by Automaat** [comment]
This pr solves the performance issue with slow creation of new dataplane on k8s with bigger deployments (600+ dpp). On deployment this size we make at most 80 qps, so default of 100 gives some headroom. This should be increased for bigger deployment, I will add docs describing this. 
Screenshot 2023-06-07 at 15 42 37

File: pkg/config/plugins/runtime/k8s/config.go [hunk 1]
```diff
@@ -120,6 +124,17 @@ type KubernetesRuntimeConfig struct {
...
 	// NodeTaintController that prevents applications from scheduling until CNI is ready.
 	NodeTaintController NodeTaintController `json:"nodeTaintController"`
+	// Kubernetes client configuration
+	ClientConfig ClientConfig `json:"clientConfig"`
+}
+
+type ClientConfig struct {
+	// Qps defines maximum requests kubernetes client is allowed to make per second.
+	// Default value 100. If set to 0 kube-client default value of 5 will be used.
+	Qps float32 `json:"qps" envconfig:"kuma_runtime_kubernetes_client_config_qps"`
```
Comment (michaelbeaumont) [inline pkg/config/plugins/runtime/k8s/config.go:134]: Should we just make this an int?

File: pkg/plugins/bootstrap/k8s/plugin.go [hunk 2]
```diff
@@ -66,6 +67,11 @@ func (p *plugin) BeforeBootstrap(b *core_runtime.Builder, cfg core_plugins.Plugi
 
 			// Disable metrics bind address as we serve metrics some other way.
 			MetricsBindAddress: "0",
+			NewClient: func(config *rest.Config, options kube_client.Options) (kube_client.Client, error) {
```
[the remark this answers is not in the document]
Comment (Automaat) [reply pkg/plugins/bootstrap/k8s/plugin.go:70]: yes, we can. I didn't catch that we pass client config there. changed

**Review by michaelbeaumont — APPROVED** [review]
:+1:

**Comment by johnharris85** [comment]
LGTM

**Pull request merged**